coherence-distributed-configuration/src/main/java/io/micronaut/coherence/discovery/CoherenceConfigurationClient.java

```diff
@@ -0,0 +1,182 @@
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.coherence.discovery;
+
+
+import com.oracle.coherence.client.GrpcSessionConfiguration;
+
+import com.tangosol.net.NamedCache;
+import com.tangosol.net.Session;
+import com.tangosol.util.Converter;
+import com.tangosol.util.Filter;
+import com.tangosol.util.Filters;
+import com.tangosol.util.ValueExtractor;
+import com.tangosol.util.extractor.KeyExtractor;
+import com.tangosol.util.function.Remote;
+
+import io.grpc.Channel;
+import io.grpc.ManagedChannelBuilder;
+
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.context.env.Environment;
+import io.micronaut.context.env.EnvironmentPropertySource;
+import io.micronaut.context.env.PropertySource;
+
+import io.micronaut.discovery.config.ConfigurationClient;
+import io.micronaut.runtime.ApplicationConfiguration;
+
+import io.reactivex.Flowable;
+import org.reactivestreams.Publisher;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Singleton
+@BootstrapContextCompatible
+public class CoherenceConfigurationClient implements ConfigurationClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CoherenceConfigurationClient.class);
+    private static final String DEFAULT_APPLICATION = "application";
+    /**
+     * The name of the map that is used to store configuration properties.
+     */
+    private static final String CONFIG_CACHE_NAME = "config-cache";
+
+    private final ApplicationConfiguration applicationConfiguration;
+    private final List<Flowable<PropertySource>> propertySources = new ArrayList<>();
+    private final CoherenceClientConfiguration coherenceClientConfiguration;
+
+    public CoherenceConfigurationClient(ApplicationConfiguration applicationConfiguration,
+                                        CoherenceClientConfiguration coherenceClientConfiguration) {
+        this.applicationConfiguration = applicationConfiguration;
+        this.coherenceClientConfiguration = coherenceClientConfiguration;
+        LOG.debug("Coherence Client configuration: " + coherenceClientConfiguration);
+    }
+
+    @Override
+    public Publisher<PropertySource> getPropertySources(Environment environment) {
+        ValueExtractor<String, String> keyExtractor = new KeyExtractor<>();
+        NamedCache<String, Object> configCache = getCache(CONFIG_CACHE_NAME, coherenceClientConfiguration);
```

Comment:
Use `NamedMap` instead -- configuration properties do not expire. Also, `configMap`, `getMap`, `CONFIG_MAP_NAME`, etc.
```
NamedMap<String, Object> configMap = getMap(CONFIG_MAP_NAME, coherenceClientConfiguration);
```

Reply:
Fixed.